lib/utils.js

```diff
@@ -689,22 +689,27 @@ function* makeCounter(seed) {
 /**
  * Helper function for either accepting a callback, or returning a promise
  *
- * @param {Function} [callback] an optional callback.
- * @param {Function} fn A function that takes a callback
+ * @param {object} parent an instance of parent with promiseLibrary.
+ * @param {object} parent.s an object containing promiseLibrary.
+ * @param {any} parent.s.promiseLibrary an object containing promiseLibrary.
```

Comment:
I think it's confusing that `parent` and `parent.s` have the same description: `an object containing promiseLibrary.`

Do we need a line for `parent.s` at all? I would think it sufficient we expect `parent` and `parent.s.promiseLibrary` to exist; the intermediate existence of parent.s being implied.

Reply:
I'm new to `jsdoc`, if we can omit that line and it still works it's no difference to me.

Reply:
when removed:

> Qualified name 'parent.s.promiseLibrary' is not allowed without a leading '@param {object} parent.s

Reply:
will changing `* @param {object} parent.s parent options object` suffice?

Reply:
How about `parent internal state` - I think that's more accurate.